meilisearch-types/src/settings.rs

```diff
@@ -532,7 +532,10 @@ pub fn settings(
 
     let faceting = FacetingSettings {
         max_values_per_facet: Setting::Set(
-            index.max_values_per_facet(rtxn)?.unwrap_or(DEFAULT_VALUES_PER_FACET),
+            index
+                .max_values_per_facet(rtxn)?
+                .map(|x| x as usize)
+                .unwrap_or(DEFAULT_VALUES_PER_FACET),
```

Comment:
Please take a look at this @dureuill

Reply:
Looks OK to me.

Reasoning below.

---

We're converting a `u64` to a `usize`.

What can go wrong?

If:
1. Ihe system is 32-bit
2. And the value is > u32::max()
Then:
- the value will be truncated

It is unclear how much support we have for (1), and (2) would mean a very large max_values_per_facet, so it is unlikely to happen in practice.

If it were to happen anyway, users would just notice a different max facet value than they specified. Which is a bug, but I think it has priority P182847.